csm/core/services/alerts.py

```diff
@@ -609,6 +609,27 @@ def stop(self):
         except Exception as e:
             Log.warn(f"Error in stopping alert monitor thread: {e}")
 
+    def _get_previous_alert(self, sensor_info, module_type):
+        """
+        This method fetches the prev alert. Before saving the alert into
+        ES DB we get the previous state of the alert.
+        During fault on the private network ES might take some time to clone
+        the data. During this duration if an alert comes CSM wont be able to
+        save it in ES and alert will be lost.
+        So, to handle this corner case when an exception comes we will sleep for
+        5 seconds and then will retry the ES connection.
+        Even if more then one alert piles up on RMQ queue this fix will handle it.
+        """
+        prev_alert = None
+        try:
+            prev_alert = self._run_coroutine\
+                (self.repo.retrieve_by_sensor_info(sensor_info, module_type))
+            return prev_alert
+        except Exception as ex:
+            Log.warn(f"Unabke to fetch previous alert.{ex}")
+            time.sleep(5)
+            self._get_previous_alert(sensor_info, module_type)
```

Comment:
Instead limit the calls to some configurable retries.

Reply:
Done